### PR TITLE
Implement API methods for admin object metadata

### PR DESCRIPTION
**Pull request by MrCreosote**
plus integration tests

### CODE REVIEW

File: src/us/kbase/workspace/kbase/ArgUtils.java [hunk 1]
```diff
@@ -407,14 +443,14 @@ public static List<Tuple7<String, String, String, Long, String, String, Long>> w
 	 * speed up the updates, but the drawback is that if the external update fails for any object,
 	 * all the objects that required updates for that system will be marked as having a failed
 	 * update. Has no effect if the permissions handler is not present.
-	 * @param logObjects if true, log the object ref and type.
+	 * @param objectInfoAsClass return the object information as a Class rather than a Tuple.
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/kbase/ArgUtils.java:446]: The answer will probably reveal itself further down the PR, but is this a new (and very sensible) feature? Is `log_objects` being silently disposed of, like the bodies of so many of your previous victims?

File: src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java [hunk 2]
```diff
@@ -1043,7 +1042,7 @@ public void saveAndGetObjects() throws Exception {
 		loi.add(new ObjectIdentity().withWsid(wsid).withName("auto2").withVer(1L));
 		loi.add(new ObjectIdentity().withWsid(wsid).withObjid(2L).withVer(1L));
 		checkSavedObjects(loi, 2, "auto2", SAFE_TYPE, 1, USER1,
-				wsid, "saveget", "36c4f68f2c98971b9736839232eb08f4", 23, meta, data);
+				wsid, "saveget", "36c4f68f2c98971b9736839232eb08f4", 23, meta, mtmeta, data);
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java:1045]: ZOMG!! How did you know that "36c4f68f2c98971b9736839232eb08f4" is my favourite string?!?!

File: src/us/kbase/workspace/kbase/ArgUtils.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/ArgUtils.java:446]: It turned out that every method that called this method set logObjects to true, so there was no need for the argument. The class vs tuple thing should be clear when you read the spec changes (you might want to do that first in fact)

File: src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java (hunk 2 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java:1045]: https://neuralink.com/

File: workspace.spec [hunk 3]
```diff
@@ -420,6 +420,42 @@ module Workspace {
 		timestamp save_date, int version, username saved_by,
 		ws_id wsid, ws_name workspace, string chsum, int size, usermeta meta>
 		object_info;
+		
+	/* Information about an object as a struct rather than a tuple.
+		This allows adding fields in a backward compatible way in the future.
+		Includes more fields than object_info.
```
Comment (ialarmedalien) [inline workspace.spec:426]: Halle-£ƒƒ1%9-lujah!!!

**Review by ialarmedalien — APPROVED** [review]
My interstitial questions were answered by the end of the PR.

**Pull request merged**